pkg/plugins/optional/grafana/v1alpha/scaffolds/edit.go

```diff
@@ -41,15 +47,122 @@ func (s *editScaffolder) InjectFS(fs machinery.Filesystem) {
+		return false
+	}
+	return true
+}
+
+func loadConfig() ([]templates.CustomMetricItem, error) {
+	// TODO: Is it good to hard code the config file path here?
+	configPath := "grafana/custom-metrics/config.yaml"
+
+	if !fileExist(configPath) {
+		return nil, nil
+	}
+
+	f, err := os.Open(configPath)
+	if err != nil {
+		return nil, fmt.Errorf("error loading plugin config: %w", err)
+	}
+
+	items, err := configReader(f)
+
+	if err := f.Close(); err != nil {
+		return nil, fmt.Errorf("could not close config.yaml: %w", err)
+	}
+
+	return items, err
+}
+
+func configReader(reader io.Reader) ([]templates.CustomMetricItem, error) {
+	yamlFile, err := ioutil.ReadAll(reader)
+	if err != nil {
+		return nil, err
+	}
+
+	config := templates.CustomMetricsConfig{}
+
+	err = yaml.Unmarshal(yamlFile, &config)
+	if err != nil {
+		return nil, err
+	}
+
+	validatedMetricItems := validateCustomMetricItems(config.CustomMetrics)
+
+	return validatedMetricItems, nil
+}
+
+func validateCustomMetricItems(rawItems []templates.CustomMetricItem) []templates.CustomMetricItem {
+	// 1. Filter items of missing `Metric` or `Type`
+	filterResult := []templates.CustomMetricItem{}
+	for _, item := range rawItems {
+		if fieldsExist(item) {
+			filterResult = append(filterResult, item)
+		}
+	}
+
+	// 2. Fill Expr if missing
+	validatedItems := make([]templates.CustomMetricItem, len(filterResult))
+	for i, item := range filterResult {
+		validatedItems[i] = fillMissingExpr(item)
+	}
+
+	return validatedItems
+}
+
+func fieldsExist(item templates.CustomMetricItem) bool {
+	// If `Expr` exists, return true
+	if item.Expr != "" {
+		return true
+	}
+
+	// If `Metric` & valid `Type` exists, return true
+	metricType := strings.ToLower(item.Type)
+	if item.Metric != "" && (metricType == "counter" || metricType == "gauge" || metricType == "histogram") {
+		return true
+	}
+
+	return false
+}
+
+func fillMissingExpr(item templates.CustomMetricItem) templates.CustomMetricItem {
+	if item.Expr == "" {
+		switch strings.ToLower(item.Type) {
+		case "counter":
+			item.Expr = "sum(rate(" + item.Metric + `{job=\"$job\", namespace=\"$namespace\"}[5m])) by (instance, pod)`
+			fmt.Println(item.Expr)
+		case "histogram":
+			// TODO: Compose a prom_ql for histogram metrics
+			// nolint: lll
+			item.Expr = "histogram_quantile(0.90, sum by(instance, le) (rate(" + item.Metric + `{job=\"$job\", namespace=\"$namespace\"}[5m])))`
+		default: // gauge
+			item.Expr = item.Metric
+		}
+	}
+	return item
+}
+
 // Scaffold implements cmdutil.Scaffolder
 func (s *editScaffolder) Scaffold() error {
 	fmt.Println("Generating Grafana manifests to visualize controller status...")
 
 	// Initialize the machinery.Scaffold that will write the files to disk
 	scaffold := machinery.NewScaffold(s.fs)
 
-	return scaffold.Execute(
+	var templatesBuilder = []machinery.Builder{
 		&templates.RuntimeManifest{},
 		&templates.ResourcesManifest{},
-	)
+		&templates.CustomMetricsConfigManifest{},
```

Comment:
Actually better option would be to take configPath as an input from here, as we do with other variables. We can let 
```go
type CustomMetricsConfigManifest {
      ConfigPath string
}
```

cc: @camilamacedo86